src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1234,28 +1236,31 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                             }
                         }
 
-                        CSharpSyntaxNode syntax = methodSymbol.GetNonNullSyntaxNode();
+                        if (!(methodSymbol is SynthesizedStaticConstructor { ShouldEmit: false }))
```

Comment:
This check is really a bit redundant, but it makes the VerifyMemberInIL calls behave as expected. Maybe we could ensure that if a method is not in "GetMethodsToEmit" we do not generate a body for it. Or we could require VerifyMemberInIL to check both the module builder and GetMethodsToEmit to decide if the "member is in IL" or not.

Reply:
> This check is really a bit redundant, but it makes the VerifyMemberInIL calls behave as expected. Maybe we could ensure that if a method is not in "GetMethodsToEmit" we do not generate a body for it. Or we could require VerifyMemberInIL to check both the module builder and GetMethodsToEmit to decide if the "member is in IL" or not.

I think that preferred way to assert that a method definition is not emitted is to use symbols. CompileAndVerify has symbolValidator exactly for that purpose. A method definition can be emitted even when it has no body, we actually need to verify that we didn't even emit a definition. Given that, I am not really concerned if VerifyMemberInIL misbehaves for this scenario, it is just a test helper and doesn't have to handle all possible scenarios. However, it does make sense to not attempt to emit the body if we are not planning to emit a declaration.

---
In reply to: [412583567](https://github.com/dotnet/roslyn/pull/43234#discussion_r412583567) [](ancestors = 412583567)